index.js

```diff
@@ -401,13 +421,20 @@ function processStyle(glStyle, map, baseUrl, host, path, accessToken) {
         glSourceId = id;
       }
       layerIds.push(glLayer.id);
+      layer.set('mapbox-source', glSourceId);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') TypeError can be thrown as "layer" might be null or undefined here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sq.boundlessgeo.io/coding_rules#rule_key=javascript%3AS2259)